spock-core/src/main/java/org/spockframework/runtime/DataIteratorFactory.java

```diff
@@ -256,6 +261,17 @@ public FeatureDataProviderIterator(IRunSupervisor supervisor, SpockExecutionCont
       dataProviders = createDataProviders();
       estimatedNumIterations = estimateNumIterations(dataProviders);
       dataProviderIterators = createDataProviderIterators();
+
+      logFilteredIterations = context
+        .getRunContext()
+        .getConfiguration(RunnerConfiguration.class)
+        .logFilteredIterations;
+      if (logFilteredIterations) {
+        stackTraceFilter = context
+          .getRunContext()
+          .getConfiguration(RunnerConfiguration.class)
+          .logFilteredIterations ? new StackTraceFilter(context.getSpec()) : new DummyStackTraceFilter();
+      }
```

Comment:
:x: Did you want to test something else here, otherwise the expressions is always true as you check the same thing in the enclosing `if`

```suggestion
      if (logFilteredIterations) {
        stackTraceFilter = context
          .getRunContext()
          .getConfiguration(RunnerConfiguration.class)
          .filterStackTrace ? new StackTraceFilter(context.getSpec()) : new DummyStackTraceFilter();
      }
```
I would also use a local var for the `config`, to avoid repeated lookups.